src/unittest/test_socket.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+Minetest
+Copyright (C) 2013 celeron55, Perttu Ahola <celeron55@gmail.com>
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation; either version 2.1 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "test.h"
+
+#include "log.h"
+#include "socket.h"
+#include "settings.h"
+
+class TestSocket : public TestBase {
+public:
+	TestSocket()
+	{
+		if (INTERNET_SIMULATOR == false)
+			TestManager::registerTestModule(this);
+	}
+
+	const char *getName() { return "TestSocket"; }
+
+	void runTests(IGameDef *gamedef);
+
+	void testIPv4Socket();
+	void testIPv6Socket();
+
+	static const int port = 30003;
+};
+
+static TestSocket g_test_instance;
+
+void TestSocket::runTests(IGameDef *gamedef)
+{
+	TEST(testIPv4Socket);
+
+	if (g_settings->getBool("enable_ipv6"))
```

Comment:
Hopefully the unit test for g_settings (is there one?) is run and working before this is called, and that g_settings is actually initialised etc etc ;) This is the reason I originally did it the way I did (i.e. this test is now dependent on a whole bunch of other tests -- if they exist -- working, and that the settings are actually initialised (which they might conceivably not be if this was a unit test without deps)